### PR TITLE
feat: integrated authentication endpoints  

### PR DESCRIPTION
**Pull request by Ifenna-Monanu**

### CODE REVIEW

File: src/components/nav/LeftNav.tsx [hunk 1]
```diff
@@ -2,51 +2,41 @@ import { IoPeopleOutline } from "react-icons/io5";
...
+}
+
+export default function LeftNavbar() {
+  const location = useLocation();
+  const pathname = location.pathname
+  return (
+    <div className="h-screen fixed w-[210px]  flex flex-col  py-5 justify-between">
+      <div className="flex flex-col px-4 py-4 gap-y-3">
+        <Navigation link="/dashboard" title="Dashboard"  icon={
+            <MdOutlineDashboard style={{color: `${pathname === '/dashboard' ? "#fff" : '#A1A1A1'}`}} />
```
Comment (Dverybest) [inline src/components/nav/LeftNav.tsx:32]: Why not create a theme file and keep all the colors there

File: src/config/index.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// import {config} from 'dotenv'
```
Comment (Dverybest) [inline src/config/index.ts:1]: what is happening here

File: src/redux/api/auth/index.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    endpoints: builder => ({
+        adminLogin: builder.mutation<ISuccessResponse<IAdmin>, ILogin>({
+            query: (data) => ({
+                url: '/login',
+                method: 'POST',
+                body: data
+            }),
+            transformErrorResponse: (
+                response: { status: number; data:IErrorResponse },
+              ) => {
```
Comment (Dverybest) [inline src/redux/api/auth/index.ts:19]: these repetitions are unnecessary, there are better ways to do this

File: src/redux/getPreloadedState.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export const getPreloadedState = () => {
+    const defaultValue = {
+        auth: {
+            user: (localStorage.getItem('adminInfo') ? JSON.parse(localStorage.getItem('adminInfo') as string) : null) as IAdmin | null,
+            access_token: (localStorage.getItem('adminToken') ? JSON.parse(localStorage.getItem('adminToken') as string) : null) as string | null
+        }
```
Comment (Dverybest) [inline src/redux/getPreloadedState.ts:6]: rewrite this  to be neat

File: src/redux/store.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (Dverybest) [inline src/redux/store.ts:1]: why this?

File: src/redux/store.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    preloadedState: getPreloadedState(),
+    reducer: combinedReducer,
+    middleware: getDefaultMiddleware => 
+        getDefaultMiddleware().concat([
+            authAPi.middleware
+        ]),
+    devTools: process.env.NODE_ENV !== 'production'   
+})
+
+export type Store = ReturnType<any>;
```
Comment (Dverybest) [inline src/redux/store.ts:25]: Remove type any

**Review by Dverybest — CHANGES_REQUESTED** [review]
why did you switch from yarn to npm?

File: src/components/Input/PasswordInput.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
+import  {useState} from 'react'
+import { useNavigate } from 'react-router-dom';
+import { ReactComponent as HideSvg } from "../../assets/icons/hide.svg";
```
[the remark this answers is not in the document]
Comment (Ifenna-Monanu) [reply src/components/Input/PasswordInput.tsx:3]: This is done in the previous PR

File: .env [hunk 8]
```diff
@@ -0,0 +1 @@
+REACT_APP_BASE_URL = https://api.tranzakt.app/docs/api/v1
```
[the remark this answers is not in the document]
Comment (Dverybest) [reply .env:1]: @Ifenna-Monanu this is still not fixed

**Review by Dverybest — CHANGES_REQUESTED** [review]

File: .env (hunk 8 above)
[the remark this answers is not in the document]
Comment (Dverybest) [reply .env:1]: this is not resolved yet

[the remark this answers is not in the document]
Comment (Ifenna-Monanu) [reply .env]: I have fixed it already thanks, please recheck

**Review by Dverybest — APPROVED** [review]

**Pull request merged**